pkgs/applications/video/avidemux/default.nix

```diff
@@ -12,122 +13,73 @@
-
-    cmakeFlags = [ "-DPLUGIN_UI=${args.pluginUi}" ];
-
-    passthru.isUi = args.isUi or false;
-
-    buildCommand = ''
-      unpackPhase
-      cd "$sourceRoot"
-      patchPhase
-
-      mkdir $out
-      lndir ${core} $out
-
-      export cmakeFlags="$cmakeFlags -DAVIDEMUX_SOURCE_DIR=$(pwd)"
-
-      for i in ${toString (args.buildDirs or [])} avidemux_plugins; do
-        ( cd "$i"
-          cmakeConfigurePhase
-          buildPhase
-          installPhase
-        )
-      done
-
-      fixupPhase
-    '';
-
-    meta = common.meta // args.meta or {};
-  });
-
-in {
-  avidemux_core = core;
-
-  avidemux_cli = buildPlugin {
-    pluginName = "cli";
-    pluginUi = "CLI";
-    isUi = true;
-    buildDirs = [ "avidemux/cli" ];
-  };
-
-  avidemux_qt4 = buildPlugin {
-    pluginName = "qt4";
-    buildInputs = [ qt4 ];
-    pluginUi = "QT4";
-    isUi = true;
-    buildDirs = [ "avidemux/qt4" ];
-  };
-
-  avidemux_gtk = buildPlugin {
-    pluginName = "gtk";
-    buildInputs = [ gtk3 ];
-    pluginUi = "GTK";
-    isUi = true;
-    buildDirs = [ "avidemux/gtk" ];
-    # Code seems unmaintained.
-    meta.broken = true;
-  };
-
-  avidemux_common = buildPlugin {
-    pluginName = "common";
-    pluginUi = "COMMON";
-  };
-
-  avidemux_settings = buildPlugin {
-    pluginName = "settings";
-    pluginUi = "SETTINGS";
+  patches = [ ./dynamic_install_dir.patch ./bootstrap_logging.patch ];
+
+  nativeBuildInputs = [ yasm cmake pkgconfig ];
+  buildInputs = [
+    zlib gettext libvdpau libva libXv sqlite fribidi fontconfig
+    freetype alsaLib libXext mesa_glu makeWrapper
+  ] ++ lib.optional withX264 x264
+    ++ lib.optional withX265 x265
+    ++ lib.optional withXvid xvidcore
+    ++ lib.optional withLAME lame
+    ++ lib.optional withFAAC faac
+    ++ lib.optional withVorbis libvorbis
+    ++ lib.optional withPulse libpulseaudio
+    ++ lib.optional withFAAD faad2
+    ++ lib.optional withOpus libopus
+    ++ lib.optionals withQT [ qttools qtbase ]
+    ++ lib.optional withVPX libvpx;
+
+  buildCommand = ''
+    unpackPhase
+    cd "$sourceRoot"
+    patchPhase
+
+    export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:${libXext}/lib"
+    ${stdenv.shell} bootStrap.bash \
+      --with-core \
+      ${if withQT then "--with-qt" else "--without-qt"} \
+      ${if withCLI then "--with-cli" else "--without-cli"} \
+      ${if withPlugins then "--with-plugins" else "--without-plugins"}
+
+    mkdir $out
+    cp -R install/usr/* $out
+
+    for i in $out/bin/*; do
+      wrapProgram $i \
+        --set ADM_ROOT_DIR $out \
+        --prefix LD_LIBRARY_PATH ":" "${libXext}/lib"
+    done
+    ln -s "$out/bin/avidemux3_${default}" "$out/bin/avidemux"
+
+    fixupPhase
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = http://fixounet.free.fr/avidemux/;
+    description = "Free video editor designed for simple video editing tasks";
+    maintainers = with maintainers; [ viric abbradar ma27 ];
```

Comment:
@viric @abbradar is it ok to keep you as maintainers as I changed way to much in this derivation :)